testcases.py

```diff
@@ -444,10 +444,17 @@ def _check_trace(self) -> bool:
             return False
 
         # check that an Initial packet uses a token sent in the Retry packet(s)
-        initials = tr.get_initial(Direction.FROM_CLIENT)
-        for p in initials:
+        highest_pn_before_retry = -1
+        for p in tr.get_initial(Direction.FROM_CLIENT):
+            pn = int(p.packet_number)
             if p.token_length == "0":
+                highest_pn_before_retry = max(highest_pn_before_retry, pn)
```

Comment:
If there are multiple Initial packets after the Retry, only the first one has to carry the token. This would break if there is reordering between the token carrying Initial and a subsequent one. Admittedly that's not going to be a problem in the tests we do, but this check isn't robust against that. Perhaps add a comment here saying so?

Reply:
I thought that the client is required to use the Retry token for all future Initial packets, but I can't find the text for that.